Source/Scene/Cesium3DTile.js

```diff
@@ -0,0 +1,1018 @@
+            }
+            var uint8Array = new Uint8Array(arrayBuffer);
+            var magic = getMagic(uint8Array);
+            var contentFactory = Cesium3DTileContentFactory[magic];
+            var content;
+
+            if (defined(contentFactory)) {
+                content = contentFactory(that._tileset, that, that._contentUrl, arrayBuffer, 0);
+                that.hasRenderableContent = true;
+            } else {
+                // The content may be json instead
+                content = Cesium3DTileContentFactory.json(that._tileset, that, that._contentUrl, arrayBuffer, 0);
+                that.hasTilesetContent = true;
+            }
+
+            that._content = content;
+            that._contentState = Cesium3DTileContentState.PROCESSING;
+            that._contentReadyToProcessPromise.resolve(content);
+
+            content.readyPromise.then(function(content) {
+                if (that.isDestroyed()) {
+                    // Tile is unloaded before the content finishes processing
+                    return when.reject('tile is destroyed');
+                }
+                updateExpireDate(that);
+
+                // Refresh style for expired content
+                that.lastStyleTime = 0;
+
+                that._contentState = Cesium3DTileContentState.READY;
+                that._contentReadyPromise.resolve(content);
+            }).otherwise(function(error) {
+                that._contentState = Cesium3DTileContentState.FAILED;
+                that._contentReadyPromise.reject(error);
+            });
+        }).otherwise(function(error) {
+            that._contentState = Cesium3DTileContentState.FAILED;
+            that._contentReadyPromise.reject(error);
+            that._contentReadyToProcessPromise.reject(error);
+        });
+
+        return true;
+    };
+
+    /**
+     * Unloads the tile's content and returns the tile's state to the state of when
+     * it was first created, before its content was loaded.
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.unloadContent = function() {
+        if (!this.hasRenderableContent) {
+            return;
+        }
+
+        this._content = this._content && this._content.destroy();
+        this._contentState = Cesium3DTileContentState.UNLOADED;
+        this._contentReadyToProcessPromise = undefined;
+        this._contentReadyPromise = undefined;
+
+        this.replacementNode = undefined;
+
+        this.lastStyleTime = 0;
+
+        this._debugColorizeTiles = false;
+
+        this._debugBoundingVolume = this._debugBoundingVolume && this._debugBoundingVolume.destroy();
+        this._debugContentBoundingVolume = this._debugContentBoundingVolume && this._debugContentBoundingVolume.destroy();
+        this._debugViewerRequestVolume = this._debugViewerRequestVolume && this._debugViewerRequestVolume.destroy();
+    };
+
+    var scratchProjectedBoundingSphere = new BoundingSphere();
+
+    function getBoundingVolume(tile, frameState) {
```

Comment:
Talked with @bagnell about this. Morphing models won't work right now since we don't store 2D and 3D positions like the regular geometries. So in following `Model.js` I stop the update early if the scene mode is morph.